PLAN.md

```diff
@@ -0,0 +1,52 @@
+- Create own directory plugin?
+    - netrw not lightweight, pollutes clipboard history
+
+- Override window navigation commands
+    - File explorer window
+        - Fixed window to the left
+        - Window navigation
+            - http://blog.paulrugelhiatt.com/vim/2014/10/31/vim-tip-automatically-create-window-splits-with-movement.html
+            - Command: `oni.window.moveLeft`, `oni.window.moveRight`, `oni.window.moveUp`, `oni.window.moveDown`
+            - WindowManager - multiple Neovim editor instances
+
+- Window focus management
+    - Create Window Manager
+    - Have focus managed
+    - Pull keyboard out externally, have focused window handle input
+        - Keyboard focus for active window
+    - Create `deactivate`, `activate` concepts for editor
+    - Push `onKeyDown` to editor
+    - How to keep from 'drilling in' to file?
```

Comment:
Sorry, some of these notes are really raw!

The 'drilling in' to the file was, when I was testing, I'd navigate a folder structure and open a file. The file would open _in the file explorer pane_ which isn't ideal. I think if we have our 'own' directory plugin, instead of netrw, we can handle this.

But basically there'd need to be a pipeline where the file-explorer-neovim handles the <enter> key, pushes it up to Oni (or maybe Oni is just using our Input work to watch the enter key), and then it would communicate to the main neovim instance to open the file. Just need to think about that flow a bit.